src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let resultState = { ...state };
+  const resultArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        for (const key in resultState) {
+          delete resultState[key];
+        }
+        resultArray.push({});
+        break;
+
+      case 'addProperties':
+        resultState = Object.assign(resultState, action.extraData);
+        resultArray.push({ ...resultState });
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          if (keyToRemove in resultState) {
+            delete resultState[keyToRemove];
+          }
+        }
+        resultArray.push({ ...resultState });
+        break;
+    }
```

Comment:
```suggestion
          delete resultState[key];
        }
        break;

      case 'addProperties':
        resultState = Object.assign(resultState, action.extraData);
        break;

      case 'removeProperties':
        for (const keyToRemove of action.keysToRemove) {
          if (keyToRemove in resultState) {
            delete resultState[keyToRemove];
          }
        }
        break;
    }
    
      resultArray.push({ ...resultState });
```

If you perform same action in all switch cases - do it just once afterwards.